data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/CsvLogstashPluginAttributesMapper.java

```diff
@@ -0,0 +1,113 @@
+                    new ArrayList<String>()
+            );
+        }
+
+        final Optional<LogstashAttribute> autodetectColumnNamesAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_AUTODETECT_COLUMN_NAMES_ATTRIBUTE_NAME);
+
+        if (logstashAttributeExistsAndIsTrue(autodetectColumnNamesAttribute)) {
+            throw new LogstashConfigurationException(LOGSTASH_AUTODETECT_COLUMN_NAMES_EXCEPTION_MESSAGE);
+        }
+
+        final Optional<LogstashAttribute> skipEmptyRowsAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_SKIP_EMPTY_ROWS_ATTRIBUTE_NAME);
+
+        if (skipEmptyRowsAttribute.isPresent()) {
+            throw new LogstashConfigurationException(LOGSTASH_SKIP_EMPTY_ROWS_EXCEPTION_MESSAGE);
+        }
+
+        final Optional<LogstashAttribute> skipEmptyColumnsAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_SKIP_EMPTY_COLUMNS_ATTRIBUTE_NAME);
+        if (skipEmptyColumnsAttribute.isPresent()) {
+            throw new LogstashConfigurationException(LOGSTASH_SKIP_EMPTY_COLUMNS_EXCEPTION_MESSAGE);
+        }
+        final Optional<LogstashAttribute> skipHeaderAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_SKIP_HEADER_ATTRIBUTE_NAME);
+        if (skipHeaderAttribute.isPresent()) {
+            throw new LogstashConfigurationException(LOGSTASH_SKIP_HEADER_EXCEPTION_MESSAGE);
+        }
+        final Optional<LogstashAttribute> convertAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_CONVERT_ATTRIBUTE_NAME);
+        if (convertAttribute.isPresent()) {
+            throw new LogstashConfigurationException(LOGSTASH_CONVERT_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private Optional<LogstashAttribute> findLogstashAttribute(final List<LogstashAttribute> logstashAttributes,
+                                                              final String logstashAttributeName) {
+        return logstashAttributes.stream()
+                .filter(logstashAttribute -> logstashAttribute.getAttributeName().equals(logstashAttributeName))
+                .findFirst();
+    }
+
+    private boolean logstashAttributeExistsAndIsTrue(final Optional<LogstashAttribute> optionalLogstashAttribute) {
+        return optionalLogstashAttribute.isPresent() && optionalLogstashAttribute.get().getAttributeValue().getValue().equals(true);
+    }
+
+    @Override
+    protected HashSet<String> getCustomMappedAttributeNames() {
+        final int numberOfCustomAttributeNames = 6;
+        final HashSet<String> names = new HashSet<>(numberOfCustomAttributeNames);
+        names.add(DATA_PREPPER_COLUMN_NAMES_SOURCE_KEY);
+        names.add(DATA_PREPPER_COLUMN_NAMES);
+        names.add(LOGSTASH_SKIP_EMPTY_ROWS_ATTRIBUTE_NAME);
+        names.add(LOGSTASH_SKIP_EMPTY_COLUMNS_ATTRIBUTE_NAME);
+        names.add(LOGSTASH_SKIP_HEADER_ATTRIBUTE_NAME);
+        names.add(LOGSTASH_CONVERT_ATTRIBUTE_NAME);
```

Comment:
I wouldn't recommend using custom mapper to throw exceptions. You can modify the existing log which lets the user know why an attribute is skipped [here](https://github.com/opensearch-project/data-prepper/blob/98ab3bbd43bc4bdf0318b52ff4a5dcfe1f822eab/data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/AbstractLogstashPluginAttributesMapper.java#L64).

Maybe something like this 
```
LOG.warn("Logstash Attribute {} is not supported in Data Prepper.", logstashAttributeName);
```